internal/jobs/launch_instance_aws.go

```diff
@@ -144,27 +144,37 @@ func DoEnsurePubkeyOnAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) e
 	fingerprint := pubkey.FindAwsFingerprint(ctx)
 	ec2Name, err := ec2Client.GetPubkeyName(ctx, fingerprint)
 	if err != nil {
-		span.SetStatus(codes.Error, "key error")
+		logger.Trace().Msgf("Pubkey with fingerprint '%s' not found, importing type %s key '%s'", fingerprint, pubkey.Type, pubkey.Body)
 
 		// if not found on AWS, import
 		if errors.Is(err, http.PubkeyNotFoundErr) {
 			pkr.Tag = ""
 			pkr.RandomizeTag()
+			retryCount := 1
+			origName := pubkey.Name
+		retry:
```

Comment:
Why do we need a sorcery here? xD
Wouldn't simple for loop with break suffice here?
I'd personally prefer it as more ppl might be more familiar with it and thus having easier time to read it :)

Reply:
Loop introduces a new indentation level, labels are feature of the language and I did some research and one of the main use across the community is for retries.

If you insist tho, I can remove and introduce a loop.